coral/src/app/features/requests/components/acls/AclRequests.test.tsx

```diff
@@ -292,6 +310,185 @@ describe("AclRequests", () => {
         expect(getAclRequests).toHaveBeenLastCalledWith({
           pageNo: "1",
           topic: "abc",
+          env: "ALL",
+          aclType: "ALL",
+          requestStatus: "ALL",
+        });
+      });
+    });
+  });
+
+  describe("user can filter ACL requests based on the environment", () => {
+    afterEach(() => {
+      cleanup();
+    });
+
+    it("populates the filter from the url search parameters", async () => {
+      mockGetEnvironments.mockResolvedValue([
+        createEnvironment({
+          name: "DEV",
+          id: "1",
+        }),
+      ]);
+
+      customRender(<AclRequests />, {
+        queryClient: true,
+        memoryRouter: true,
+        customRoutePath: "/?environment=1",
+      });
+
+      await waitForElementToBeRemoved(
+        screen.getByTestId("select-environment-loading")
+      );
+
+      const envFilter = screen.getByRole("combobox", {
+        name: "Filter by Environment",
+      });
+
+      expect(envFilter).toHaveDisplayValue("DEV");
+
+      expect(getAclRequests).toHaveBeenNthCalledWith(1, {
+        pageNo: "1",
+        topic: "",
+        env: "1",
+        aclType: "ALL",
+        requestStatus: "ALL",
+      });
+    });
+
+    it("applies the topic filter by selecting a value", async () => {
```

Comment:
That's the filter for environments. :D 

"topic filter" appears multiple time and needs to be updated. It could be "ACL filter" but I personally would be more specific per test case, aka "enables user to filter request by Environment" - it's more precise and does describe the test case from user perspective, not from the implementation detail.

Reply:
Arggh I thought I had found all the c/c mistakes!

Reply:
[31b2161](https://github.com/aiven/klaw/pull/852/commits/31b2161f77e72b4ac5e30fbec46b1d337c849ae9)